src/pymor/algorithms/error.py

```diff
@@ -257,56 +264,121 @@ def custom_value(rom, fom, reductor, mu, dim):
     result['summary'] = summary
 
     if plot:
-        import matplotlib.pyplot as plt
-        fig = plt.figure()
-        num_plots = (int(bool(error_norms) or error_estimator) + int(bool(error_norms) and error_estimator)
-                     + int(condition) + int(bool(custom)))
-        current_plot = 1
-
-        if bool(error_norms) or error_estimator:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            legend = []
-            if error_norms:
-                for name, errors in zip(error_norm_names, max_errors):
-                    ax.semilogy(basis_sizes, errors)
-                    legend.append(name)
-            if error_estimator:
-                ax.semilogy(basis_sizes, max_error_estimates)
-                legend.append('error estimator')
-            ax.legend(legend)
-            ax.set_title('maximum errors')
-            current_plot += 1
-
-        if bool(error_norms) and error_estimator:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            ax.semilogy(basis_sizes, min_effectivities)
-            ax.semilogy(basis_sizes, max_effectivities)
-            ax.legend(('min', 'max'))
-            ax.set_title('error estimator effectivities')
-            current_plot += 1
+        result['figure'] = plot_reduction_error_analysis(
+                result, None, plot_custom_logarithmic=plot_custom_logarithmic, return_fig=True)
 
-        if condition:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            ax.semilogy(basis_sizes, max_conditions)
-            ax.set_title('maximum condition')
-            current_plot += 1
-
-        if custom:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            legend = []
-            for i, values in enumerate(custom_values):
-                if plot_custom_logarithmic:
-                    ax.semilogy(basis_sizes, values)
-                else:
-                    ax.plot(basis_sizes, values)
-                legend.append('value ' + str(i))
-            ax.legend(legend)
-            ax.set_title('maximum custom values')
-            current_plot += 1
+    return result
 
-        result['figure'] = fig
 
-    return result
+
+def plot_reduction_error_analysis(result, max_basis_size=None, plot_effectivities=True, plot_condition=True,
```

Comment:
I don't understand why returning the figure is optional. Can you help me with that?

Reply:
Just to have the same behavior as before. Open for suggestions ...

Reply:
But the function is always called with `return_fig=True` and the snippet it was refactored from always put the figure into the result?
The result dict might not have had a figure if `plot == False`, but that is not what you wanted to express here?

Reply:
> But the function is always called with `return_fig=True` and the snippet it was refactored from always put the figure into the result?

No, its not. Usually, I call
```python
data = reduction_error_analysis(...)

plot_reduction_error_analysis(data, ...)
```
which leaves `return_fig=False`.

> The result dict might not have had a figure if `plot == False`, but that is not what you wanted to express here?

I guess we can argue about if we want to have
```python
plot_reduction_error_analysis(..., return_fig=False)
```
or
```python
_ = plot_reduction_error_analysis(...)
```
However, I would like to keep the
```python
if plot:
    result['figure'] = ...
```
part as is, since it allows to collect the data without plotting to work.

Reply:
Right, I was not precise. "Always" was really only from pyMOR itself, did not consider the outside context. 
So basically you want to configure the return value from either `None` or `figure` so that you can suppress automatic (inline) display in a notebook/ipython context?

Reply:
I guess. There are historical reasons. Since the `figure` was present in the `result` data, I decided to let the `plot_...` return the figure and started to adapt later. I really don't care much about the details. I just want to collect the data without plotting and want a helper to plot in one function without knowing about matplotlib internals. I am open to suggestions...

Reply:
Any other thoughts here, @sdrave maybe?

Reply:
I think if we have seperate methods for computing the data and plotting, then I would remove the figure from `results`. I'm fine with having `return_fig` for convenience, but I don't have a strong opinion on that.

> However, I would like to keep the
> ```python
> if plot:
>     result['figure'] = ...
> ```
> part as is, since it allows to collect the data without plotting to work.

Why doesn't that work without the plot argument?!

Reply:
Currently, with `plot` being `True`, the `reduction_error_analysis` calls `plot_reduction_error_analysis`, where errors might occur for broken matplotlib backends. Bu nvm that. Should we then remove the `plot` argument from `reduction_error_analysis` altogether, s.t. the used would always have to call both methods, if interested in a plot?

Reply:
I left the `plot` (could also be dropped IMHO), and do not return the figure.